### ISSUE TITLE
☂️ Issue Checklist for GRPC v1.0

### ISSUE THREAD

**Issue by glbrntt** [grpc/grpc-swift#492]
This is an umbrella issue with a list of things to do before we can release 1.0.0.

- [x] Rework the connectivity delegate API
- [x] Rename the `GRPCNIO` helper enum
- [ ] Add logging with [swift-log](https://github.com/apple/swift-log)
  - [x] Client
  - [ ] Server
- [ ] Use the same `EventLoop` in `ClientConnection` after a reconnect (depends on: https://github.com/apple/swift-nio-transport-services/pull/49)
- [x] Make TLS easier to configure
- [ ] A guide to help users transition from the old implementation
- [ ] Announce the project in the README on `master`
- [ ] Provide CocoaPods support

**Comment by glbrntt** [comment]
Closing in favour of #526
